ecl/hthor/hthor.cpp

```diff
@@ -372,13 +375,19 @@ ClusterWriteHandler *createClusterWriteHandler(IAgentContext &agent, IHThorIndex
         }
         if (!cluster)
             break;
-        if(!clusterHandler)
+        if (!clusterHandler)
         {
-            if(extend)
-                throw MakeStringException(0, "Cannot combine EXTEND and CLUSTER flags on disk write of file %s", lfn);
+            if (extend)
+                throw makeStringExceptionV(ENGINEERR_EXTEND_CLUSTER_WRITE, "Cannot combine EXTEND and CLUSTER flags on disk write of file %s", lfn);
             clusterHandler.setown(new CHThorClusterWriteHandler(lfn, "OUTPUT", agent));
         }
         clusterHandler->addCluster(cluster);
+        Owned<IPropertyTree> plane = getStoragePlane(cluster);
+        bool thisPlaneCompressed = plane ? plane->getPropBool("@compressLogicalFiles") : false;
+        if (1 == clusterIdx)
+            outputPlaneCompressed = thisPlaneCompressed;
+        else if (outputPlaneCompressed != thisPlaneCompressed)
+            throw makeStringExceptionV(ENGINEERR_MIXED_COMPRESSED_WRITE, "Writing to planes with mixed compression is not supported %s", lfn);
```

Comment:
Is this a helpful error, or should it just use the compression from the 1st plane?  The option on the plane only provides the default.

Reply:
could do, I think this is unlikely to be hit, because we don't support multiple plane output in cloud in other places at the moment (e.g. : https://github.com/hpcc-systems/HPCC-Platform/blob/a93c4070959c56d5e32c07398ce9677495de1d49/common/thorhelper/roxiehelper.cpp#L2467)

But, I thought it better to check and throw an early error, rather than continue, and at some later point cause confusion.